rust-runtime/aws-smithy-http/src/header.rs

```diff
@@ -560,4 +618,28 @@ mod test {
         assert_eq!("\"(\"", &quote_header_value("("));
         assert_eq!("\")\"", &quote_header_value(")"));
     }
+
+    #[test]
+    fn test_append_merge_header_maps() {
```

Comment:
There should also be a test for multiple header values with the same header name in the same header map. Probably also a good idea to have tests with empty header maps.